src/backup.ts

```diff
@@ -0,0 +1,385 @@
+   * // Await completion of the backup operation.
+   * await backupOperation.promise();
+   */
+  create(
+    callback?: CreateBackupCallback
+  ): Promise<CreateBackupResponse> | void {
+    if (!this.expireTime) {
+      throw new Error('Expire time is required to create a backup.');
+    }
+    if (!this.databasePath) {
+      throw new Error('Database path is required to create a backup.');
+    }
+
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.ICreateBackupRequest = extend(
+      {
+        parent: this.instance.formattedName_,
+        backupId: this.backupId,
+        backup: {
+          database: this.databasePath,
+          expireTime: this.expireTime.toStruct(),
+          name: this.formattedName_,
+        },
+      }
+    );
+    return this.request<CreateBackupGaxOperation>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'createBackup',
+        reqOpts,
+      },
+      (err, resp) => {
+        if (err) {
+          callback!(err, null, null);
+          return;
+        }
+        callback!(null, this, resp, resp || undefined);
+      }
+    );
+  }
+
+  getMetadata(): Promise<GetMetadataResponse>;
+  getMetadata(callback: GetMetadataCallback): void;
+  /**
+   * @typedef {array} GetMetadataResponse
+   * @property {object} 0 The {@link Backup} metadata.
+   * @property {object} 1 The full API response.
+   */
+  /**
+   * @callback GetMetadataCallback
+   * @param {?Error} err Request error, if any.
+   * @param {object} metadata The {@link Backup} metadata.
+   * @param {object} apiResponse The full API response.
+   */
+  /**
+   * Retrieves backup's metadata.
+   *
+   * @see {@link #getState}
+   * @see {@link #getExpireTime}
+   *
+   * @method Backup#getMetadata
+   * @returns {Promise<GetMetadataResponse>}
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const database = spanner.database('my-database');
+   * const backup = instance.backup('my-backup');
+   * const [backupInfo] = await backup.getMetadata();
+   * console.log(`${backupInfo.name}: size=${backupInfo.sizeBytes}`);
+   */
+  getMetadata(
+    callback?: GetMetadataCallback
+  ): void | Promise<GetMetadataResponse> {
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.IGetBackupRequest = {
+      name: this.formattedName_,
+    };
+    return this.request<IBackupTranslatedEnum>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'getBackup',
+        reqOpts,
+      },
+      (err, response) => {
+        callback!(err, response);
+      }
+    );
+  }
+
+  /**
+   * Retrieves the state of the backup.
+   *
+   * The backup state indicates if the backup has completed.
+   *
+   * @see {@link #getMetadata}
+   *
+   * @method Backup#getState
+   * @returns {Promise<EnumKey<typeof databaseAdmin.spanner.admin.database.v1.Backup.State> | undefined>}
```

Comment:
@mayurkale22 is there some other way to represent this type? The docs presubmit is unhappy with this but I'm not sure how to fix it.

```
ERROR: Unable to parse a tag's type expression for source file /usr/local/google/home/skuruppu/src/cloud_spanner_clients/nodejs-spanner-skuruppu/build/src/backup.js in line 131 with tag title "returns" and text "{Promise<EnumKey<typeof databaseAdmin.spanner.admin.database.v1.Backup.State> | undefined>}
    when resolved, contains the current state of the backup if it exists, or
    undefined if the backup does not exist.": Invalid type expression "Promise<EnumKey<typeof databaseAdmin.spanner.admin.database.v1.Backup.State> | undefined>": Expected ",", ">", or "|" but "d" found.
ERROR: Unable to parse a tag's type expression for source file /usr/local/google/home/skuruppu/src/cloud_spanner_clients/nodejs-spanner-skuruppu/build/src/backup.js in line 151 with tag title "returns" and text "{Promise<EnumKey<typeof databaseAdmin.spanner.admin.database.v1.Backup.State> | undefined>}
    when resolved, contains the current state of the backup if it exists, or
    undefined if the backup does not exist.": Invalid type expression "Promise<EnumKey<typeof databaseAdmin.spanner.admin.database.v1.Backup.State> | undefined>": Expected ",", ">", or "|" but "d" found.
```

Reply:
Option1: Use `@typedef`: https://jsdoc.app/tags-typedef.html
```
/*
 * .....
 * @method Backup#getState
 * @typedef {databaseAdmin.spanner.admin.database.v1.Backup.State} State
 * @returns {Promise<EnumKey<State> | undefined>}
 *     when resolved, contains the current state of the backup if it exists, or
 *     undefined if the backup does not exist.
 * ....
 */
```

Option2: Use `,` as per the suggestion in error (Expected ",", ">", or "|" but "d" found)
```
/*
 * .....
 * @method Backup#getState
 * @returns {Promise<EnumKey<typeof, databaseAdmin.spanner.admin.database.v1.Backup.State> | undefined>}
 *     when resolved, contains the current state of the backup if it exists, or
 *     undefined if the backup does not exist.
 * ....
 */
```
Although `npm run docs-test` is passing for both the options,  `getState()` is not visible in the documentation when I use Option 1. That's why I would suggest to you Option 2.

Reply:
Interesting, Option1 looks cleaner but I agree, it doesn't appear in the docs. I went with Option2 as you suggested.